crates/biome_js_analyze/src/lint/nursery/use_jsx_curly_brace_convention.rs

```diff
@@ -0,0 +1,339 @@
+use biome_analyze::{
+    context::RuleContext, declare_rule, ActionCategory, Ast, FixKind, Rule, RuleDiagnostic,
+    RuleSource, RuleSourceKind,
+};
+use biome_console::markup;
+use biome_js_factory::make;
+use biome_js_syntax::{
+    AnyJsExpression, AnyJsLiteralExpression, AnyJsxAttributeValue, AnyJsxChild, JsSyntaxKind,
+    JsSyntaxToken, JsxAttributeInitializerClause, JsxExpressionAttributeValue, T,
+};
+use biome_rowan::{declare_node_union, AstNode, BatchMutationExt, TextRange};
+
+use crate::JsRuleAction;
+
+declare_rule! {
+    /// This rule allows you to enforce curly braces or disallow unnecessary curly braces in JSX props and/or children.
+    ///
+    /// For situations where JSX expressions are unnecessary, please refer to [the React doc](https://facebook.github.io/react/docs/jsx-in-depth.html) and [this page about JSX gotchas](https://github.com/facebook/react/blob/v15.4.0-rc.3/docs/docs/02.3-jsx-gotchas.md#html-entities).
+    ///
+    /// By default, this rule will check for and warn about unnecessary curly braces in both JSX props and children.
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    /// ```jsx,expect_diagnostic
+    /// <Foo>{'Hello world'}</Foo>;
+    ///
+    /// <Foo foo={'bar'} />;
+    ///
+    /// <Foo foo=<Bar /> />;
```

Comment:
This is an invalid code snippet. Multiple JSX elements needs to be wrapped inside a Fragment. Plus, you should not have trailing `;`, because it's actually text:

```suggestion
    /// <>
    ///   <Foo>{'Hello world'}</Foo>
    ///   <Foo foo={'bar'} />
    ///   <Foo foo=<Bar /> />
    /// </>
```

I believe this suggestion will fail because it will *multiple diagnostics*, which is invalid. Each snippet must [emit one diagnostic](https://github.com/biomejs/biome/blob/main/crates/biome_analyze/CONTRIBUTING.md#document-the-rule):

> When adding invalid snippets in the ### Invalid section, you must use the expect_diagnostic code block property. We use this property to generate a diagnostic and attach it to the snippet. A snippet must emit only ONE diagnostic.